source/server/http/admin.cc

```diff
@@ -661,15 +662,21 @@ envoy::admin::v2alpha::ServerInfo::State AdminImpl::serverState() {
 
 Http::Code AdminImpl::handlerServerInfo(absl::string_view, Http::HeaderMap& headers,
                                         Buffer::Instance& response, AdminStream&) {
-  time_t current_time = time(nullptr);
+  const auto current_time = server_.timeSource().systemTime();
   envoy::admin::v2alpha::ServerInfo server_info;
   server_info.set_version(VersionInfo::version());
   server_info.set_state(serverState());
 
-  server_info.mutable_uptime_current_epoch()->set_seconds(current_time -
-                                                          server_.startTimeCurrentEpoch());
-  server_info.mutable_uptime_all_epochs()->set_seconds(current_time -
-                                                       server_.startTimeFirstEpoch());
+  ASSERT(current_time > server_.startTimeCurrentEpoch());
+  const uint64_t current_uptime = std::chrono::duration_cast<std::chrono::seconds>(
+                                      current_time - server_.startTimeCurrentEpoch())
```

Comment:
Should we check for server_.startTimeCurrentEpoch() > current_time before doing this subtraction, here and the other places in the PR where that subtraction is done.